lib/encryptonize_test.go

```diff
@@ -61,3 +61,709 @@ func TestEncryptDecrypt(t *testing.T) {
+// 3) It is verified that only user1 who is part of the access object is able to call GetAccessGroups.
+func TestGetAccessGroups(t *testing.T) {
+	enc := newTestEncryptonize(t)
+
+	user1, _, _, err := enc.NewUser(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	user2, _, _, err := enc.NewUser(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	plainObject := Object{
+		Plaintext:      []byte("plaintext"),
+		AssociatedData: []byte("associated_data"),
+	}
+
+	_, access, err := enc.Encrypt(&user1, &plainObject)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = enc.GetAccessGroups(&user1, &access)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = enc.GetAccessGroups(&user2, &access)
+	if err == nil {
+		t.Fatal(err)
+	}
+}
+
+// Scenario:
+// 1) Two users are created, user1 and user2.
+// 2) user1 encrypts an object.
+// 3) It is verified that the group with same ID as user1 can be added and removed from the access object.
+// 4) It is verified that only user1 who is part of the access object is able to add and remove the group.
+func TestAddRemoveGroupsFromAccess(t *testing.T) {
+	enc := newTestEncryptonize(t)
+
+	user1, group, _, err := enc.NewUser(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	user2, _, _, err := enc.NewUser(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	plainObject := Object{
+		Plaintext:      []byte("plaintext"),
+		AssociatedData: []byte("associated_data"),
+	}
+
+	_, access, err := enc.Encrypt(&user1, &plainObject)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = enc.AddGroupsToAccess(&user2, &access, &group)
+	if err == nil {
+		t.Fatal(err)
+	}
+
+	err = enc.AddGroupsToAccess(&user1, &access, &group)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	accessGroups, err := enc.GetAccessGroups(&user1, &access)
+
+	if _, ok := accessGroups[group.ID]; !ok {
+		t.Fatal("Group not correctly added to access")
+	}
+
+	err = enc.RemoveGroupsFromAccess(&user2, &access, &group)
+	if err == nil {
+		t.Fatal(err)
+	}
+
+	err = enc.RemoveGroupsFromAccess(&user1, &access, &group)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	accessGroups, err = enc.GetAccessGroups(&user1, &access)
+
+	if _, ok := accessGroups[group.ID]; ok {
+		t.Fatal("Group not correctly removed from access")
+	}
+}
+
+// Scenario:
+// 1) Two users are created, user1 and user2.
+// 2) user1 encrypts an object.
+// 3) It is verified that only user1 is authorized.
+// 4) It is verified that if user1's ciphertext is changed, then user1 is not authorized anymore.
+func TestAuthorizeUser(t *testing.T) {
+	enc := newTestEncryptonize(t)
+
+	user1, _, _, err := enc.NewUser(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	user2, _, _, err := enc.NewUser(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	plainObject := Object{
+		Plaintext:		[]byte("plaintext"),
+		AssociatedData:	[]byte("associated_data"),
+	}
+
+	_, access, err := enc.Encrypt(&user1, &plainObject)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = enc.AuthorizeUser(&user1, &access)
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
For stuff like this, you can also use the following syntax (might also be useful in other places in this file):
```suggestion
	if err := enc.AuthorizeUser(&user1, &access); err != nil {
		t.Fatal(err)
	}
```